library/core/src/alloc/global.rs

```diff
@@ -221,20 +222,21 @@ pub unsafe trait GlobalAlloc {
     ///
     /// # Safety
     ///
-    /// This function is unsafe because undefined behavior can result
-    /// if the caller does not ensure all of the following:
+    /// The caller has to ensure that
     ///
-    /// * `ptr` must be currently allocated via this allocator,
+    /// * `ptr` is currently allocated via this allocator,
     ///
-    /// * `layout` must be the same layout that was used
+    /// * `layout` is the same layout that was used
     ///   to allocate that block of memory,
     ///
-    /// * `new_size` must be greater than zero.
+    /// * `new_size` is greater than zero.
     ///
     /// * `new_size`, when rounded up to the nearest multiple of `layout.align()`,
-    ///   must not overflow isize (i.e., the rounded value must be less than or
+    ///   does not overflow isize (i.e., the rounded value must be less than or
     ///   equal to `isize::MAX`).
     ///
+    /// Otherwise undefined behaviour can result.
```

Comment:
```suggestion
    /// If these are not followed, undefined behaviour can result.
```

Just clarify what `otherwise` refers to.